app/org/sagebionetworks/bridge/services/StudyService.java

```diff
@@ -225,19 +239,30 @@ public Study createSynapseProjectTeam(Long userId, Study study) throws SynapseEx
 
     public Study updateStudy(Study study, boolean isAdminUpdate) {
         checkNotNull(study, Validate.CANNOT_BE_NULL, "study");
-        
+
         sanitizeHTML(study);
 
         // These cannot be set through the API and will be null here, so they are set on update
         Study originalStudy = studyDao.getStudy(study.getIdentifier());
+
         study.setStormpathHref(originalStudy.getStormpathHref());
-        study.setActive(true);
-        // And this cannot be set unless you're an administrator. Regardless of what the 
+        // And this cannot be set unless you're an administrator. Regardless of what the
         // developer set, set these back to the original study.
         if (!isAdminUpdate) {
+            // prevent non-admins update a deactivated study
+            if (!originalStudy.isActive()) {
+                throw new EntityNotFoundException(Study.class, "Study '"+ study.getIdentifier() +"' is deactivated.");
```

Comment:
Study not found (don't use word deactivated in API). Also if you're going to throw a 404, you might as well do it before doing any further work, so I would get the original study and check it at the top of the method. Same with the exception you throw below.